tests/Unit/PointwiseFunctions/AnalyticSolutions/Elasticity/Test_HalfSpaceMirror.cpp

```diff
@@ -0,0 +1,122 @@
+  }
+
+  // NOLINTNEXTLINE(readability-convert-member-functions-to-static)
+  tuples::tagged_tuple_from_typelist<source_tags> source_variables(
+      const tnsr::I<DataVector, 3>& x) const noexcept {
+    return Elasticity::Solutions::HalfSpaceMirror::variables(x, source_tags{});
+  }
+};
+
+}  // namespace
+
+SPECTRE_TEST_CASE(
+    "Unit.PointwiseFunctions.AnalyticSolutions.Elasticity.HalfSpaceMirror",
+    "[PointwiseFunctions][Unit][Elasticity]") {
+  const size_t dim = Elasticity::Solutions::HalfSpaceMirror::dim;
+  const Elasticity::Solutions::HalfSpaceMirror check_solution{
+      0.177, 1.0,
+      Elasticity::ConstitutiveRelations::IsotropicHomogeneous<3>{
+          // Iron: E=72, nu=0.17
+          36.36363636363637, 30.76923076923077},
+      50, 1.e-13};
+  const auto created_solution =
+      TestHelpers::test_creation<Elasticity::Solutions::HalfSpaceMirror>(
+          "BeamWidth: 0.177\n"
+          "AppliedForce: 1.0\n"
+          "Material:\n"
+          "  BulkModulus: 36.36363636363637\n"
+          "  ShearModulus: 30.76923076923077\n"
+          "IntegrationIntervals: 20\n"
+          "IntergrationTolerance: 1.e-13\n");
+  CHECK(created_solution == check_solution);
+  test_serialization(check_solution);
+
+  pypp::SetupLocalPythonEnvironment local_python_env{"PointwiseFunctions"};
+  const Elasticity::ConstitutiveRelations::IsotropicHomogeneous<dim>
+      constitutive_relation{36.36363636363637, 30.76923076923077};
+  const HalfSpaceMirrorProxy solution{0.177, 1.0, constitutive_relation, 100,
+                                      1.e-13};
+  pypp::check_with_random_values<1,
+                                 tmpl::list<Elasticity::Tags::Displacement<dim>,
+                                            Elasticity::Tags::Strain<dim>>>(
+      &HalfSpaceMirrorProxy::field_variables, solution,
+      "AnalyticSolutions.Elasticity.HalfSpaceMirror",
+      {"displacement", "strain"}, {{{0., 3.}}},
+      std::make_tuple(0.177, 1.0, 36.36363636363637, 30.76923076923077),
+      DataVector(5));
+  pypp::check_with_random_values<
+      1, tmpl::list<Tags::FixedSource<Elasticity::Tags::Displacement<dim>>>>(
+      &HalfSpaceMirrorProxy::source_variables, solution,
+      "AnalyticSolutions.Elasticity.HalfSpaceMirror", {"source"}, {{{0., 3.}}},
+      std::make_tuple(), DataVector(5));
+
+  {
+    INFO("Test elasticity system with half-space mirror");
+    // Verify that the solution numerically solves the system and that the
+    // discretization error decreases exponentially with polynomial order
+    using system = Elasticity::FirstOrderSystem<dim>;
+    const typename system::fluxes fluxes_computer{};
+    using AffineMap = domain::CoordinateMaps::Affine;
+    using AffineMap3D =
+        domain::CoordinateMaps::ProductOf3Maps<AffineMap, AffineMap, AffineMap>;
+    const domain::CoordinateMap<Frame::Logical, Frame::Inertial, AffineMap3D>
+        coord_map{{{-1., 1., 0., 0.5}, {-1., 1., 0., 0.5}, {-1., 1., 0., 0.5}}};
```

Comment:
Perhaps try making the domain a bit smaller (maybe 0.3) since the beam width is only ~0.2 to get good convergence on a single element with just a few points.

Reply:
I'm going through all the messages. I didn't mean to gloss over this message. Do you think the convergence for the test is good as is or is this still relevant?